exercises/anagram/test/Tests.hs

```diff
@@ -1,40 +1,140 @@
-  , testCase "detect anagram" $
-    ["inlets"] @=? anagramsFor "listen" ["enlists", "google",
-                                         "inlets", "banana"]
-  , testCase "multiple anagrams" $
-    ["gallery", "regally", "largely"] @=?
-    anagramsFor "allergy" ["gallery", "ballerina", "regally", "clergy",
-                           "largely", "leading"]
-  , testCase "case insensitive anagrams" $
-    ["Carthorse"] @=?
-    anagramsFor "Orchestra" ["cashregister", "Carthorse", "radishes"]
-  , testCase "does not detect a word as its own anagram" $
-    [] @=? anagramsFor "banana" ["banana"]
-  , testCase "does not detect a word as its own anagram (case insensitive)" $
-    [] @=? anagramsFor "Banana" ["baNana"]
-  ]
+main = hspecWith defaultConfig {configFastFail = True} specs
+
+specs :: Spec
+specs = describe "anagram" $
+          describe "anagramsFor" $ for_ cases test
+  where
+
+    test Case{..} = it description $ expression `shouldBe` expected
+      where
+        expression = anagramsFor subject candidates
+
+-- Test cases adapted from `exercism/x-common/anagrams.json` on 2016-07-25.
+
+data Case = Case { description ::  String
+                 , subject     ::  String
+                 , candidates  :: [String]
+                 , expected    :: [String]
+                 }
+
+cases :: [Case]
+cases = [ Case { description = "no matches"
+               , subject     = "diaper"
+               , candidates  = [ "hello", "world", "zombies", "pants"]
+               , expected    = []
+               }
+        , Case { description = "detects simple anagram"
+               , subject     = "ant"
+               , candidates  = ["tan", "stand", "at"]
+               , expected    = ["tan"]
+               }
+        , Case { description = "does not detect false positives"
+               , subject     = "galea"
+               , candidates  = ["eagle"]
+               , expected    = []
+               }
+        , Case { description = "detects multiple anagrams"
+               , subject     = "master"
+               , candidates  = ["stream", "pigeon", "maters"]
+               , expected    = ["stream", "maters"]
+               }
+        , Case { description = "does not detect anagram subsets"
+               , subject     = "good"
+               , candidates  = ["dog", "goody"]
+               , expected    = []
+               }
+        , Case { description = "detects anagram"
+               , subject     = "listen"
+               , candidates  = ["enlists", "google", "inlets", "banana"]
+               , expected    = ["inlets"]
+               }
+        , Case { description = "detects multiple anagrams"
+               , subject     = "allergy"
+               , candidates  = ["gallery", "ballerina", "regally", "clergy", "largely", "leading"]
+               , expected    = ["gallery", "regally", "largely"]
+               }
+        , Case { description = "does not detect indentical words"
```

Comment:
I see the spelling mistake indentical is in x-common as well, remind me to file a PR there as well...